terraform/modules/networking/subnet.tf

```diff
@@ -105,11 +106,21 @@ resource "aws_subnet" "github_runner" {
   tags                    = { Name = "github-runner-private-subnet-${var.environment}" }
 }
 
-resource "aws_subnet" "vpc_endpoints_subnet" {
-  availability_zone = data.aws_availability_zones.available.names[0]
-  cidr_block        = local.vpc_endpoints_cidr_8
+resource "aws_subnet" "cpm_private" {
+  count                   = 3
+  cidr_block              = cidrsubnet(local.cpm_private_cidr_10, 2, count.index)
+  vpc_id                  = aws_vpc.vpc.id
+  availability_zone       = data.aws_availability_zones.available.names[count.index]
+  map_public_ip_on_launch = false
+  tags                    = { Name = "cpm-private-subnet-${data.aws_availability_zones.available.names[count.index]}-${var.environment}" }
+}
+
+resource "aws_subnet" "vpc_endpoints_subnets" {
+  count             = 3
```

Comment:
@BenRamchandani I know you said you don't think a subnet per AZ is actually necessary, but the docs recommended it and I don't see the downside?

Reply:
Agree we should do it, but assuming I'm reading this correctly it does cost money, <https://aws.amazon.com/privatelink/pricing/>.

```text
0.011$/hour * 720 hours/month * 2 AZs * 12 endpoints = 190.08$/month
```

So about £170/month for each environment. Fine for production and staging should match, not sure whether it's worth the extra complexity of going down to one subnet to make test a bit cheaper.